contrib/seeds/makeseeds.py

```diff
@@ -179,13 +155,14 @@ def filterbyasn(ips: List[Dict], max_per_asn: Dict, max_per_net: int) -> List[Di
             # do not add this ip as we already too many
             # ips from this network
             continue
-        asn = lookup_asn(ip['net'], ip['ip'])
-        if asn is None or asn_count[asn] == max_per_asn[ip['net']]:
+        asn = asmap.lookup_asn(ip['ip'])
+        if asn is None or asn_count[ip['net'], asn] == max_per_asn[ip['net']]:
             # do not add this ip as we already have too many
             # ips from this ASN on this network
             continue
-        asn_count[asn] += 1
+        asn_count[ip['net'], asn] += 1
```

Comment:
btw: This fixes a bug that was exposed by the changes in #24818, where it was possible for a limit to be exceeded because `max_per_asn` differs per network but `asn_count` was not counted per network (in combination with `==`). Making the counting per ipv4/ipv6 avoids this, and makes sense to me conceptually. So after this, a single ASN can have *both* 2 IPv4 seeds and 6 IPv6 seeds.

Reply:
With this change I only see 12 entries in nodes_main.txt though (2 IPv4, 10 IPv6, all AS30765).

Reply:
Which asmap are you using? It seems more likely to me that your asmap is wrong, or something is wrong with parsing, causing all IPs to be assigned to one AS, than that this change is at fault.

Reply:
FWIW this is my map file:
```
sha256sum demo.map
94b6142b574d468e73c71ed9fb411683d295c37b756f1f735013f736ee647330  demo.map
```
Output:
```
Loading asmap database demo.map... Done.
  IPv4   IPv6  Onion Pass
470789  73296      0 Initial
470789  73296      0 Skip entries with invalid address
470789  73296      0 After removing duplicates
  6318   1747      0 Enforce minimal number of blocks
  5456   1504      0 Require service bit 1
  3866    887      0 Require minimum uptime
  3789    864      0 Require a known and recent user agent
  3760    856      0 Filter out hosts with multiple bitcoin ports
   512    301      0 Look up ASNs and limit results per ASN and per net
```

Reply:
I currently am having this same bug using the asmap from the readme.md included.
`curl https://github.com/sipa/asmap/raw/master/demo.map > demo.map`

```
sha256sum demo.map
72034b228f2b91d8c0864d27d5f8cbbad4b34f1b0948fd2bc9f1acf66aaf168c  demo.map
```
```
Loading asmap database demo.map... Done.
  IPv4   IPv6  Onion Pass
470888  73307      0 Initial
470888  73307      0 Skip entries with invalid address
470888  73307      0 After removing duplicates
470887  73307      0 Skip entries from suspicious hosts
  6502   1787      0 Enforce minimal number of blocks
  5610   1523      0 Require service bit 1
  3862    896      0 Require minimum uptime
  3777    869      0 Require a known and recent user agent
  3753    861      0 Filter out hosts with multiple bitcoin ports
     2     10      0 Look up ASNs and limit results per ASN and per net

real    0m11.036s
user    0m10.079s
sys     0m0.907s
```

The time to parse is insanely fast but the included file is only 2.5kb which to me seems incredibly small as an uncompressed asmap is nearly 27MB.

Reply:
@russeree If you use curl, it'll just store "You are being redirected to ..." in the file.

That demo.dap file is also very old, and wasn't even intended back then to be complete. Try https://bitcoin.sipa.be/asmap-filled.dat.

Reply:
```
cat demo.map
<html><body>You are being <a href="https://raw.githubusercontent.com/sipa/asmap/master/demo.map">redirected</a>.</body></html>
```

well then ... after using the correct file things looked much better.

`Loading asmap database demo.map... Done.
  IPv4   IPv6  Onion Pass
470888  73307      0 Initial
470888  73307      0 Skip entries with invalid address
470888  73307      0 After removing duplicates
470887  73307      0 Skip entries from suspicious hosts
  6502   1787      0 Enforce minimal number of blocks
  5610   1523      0 Require service bit 1
  3862    896      0 Require minimum uptime
  3777    869      0 Require a known and recent user agent
  3753    861      0 Filter out hosts with multiple bitcoin ports
   512    291      0 Look up ASNs and limit results per ASN and per net

real    0m15.295s
user    0m14.423s
sys     0m0.808s`

using a broken file presents an interesting note to me. When using the broken file the script just used "You are being redirected to ..." as the ASN for ipv4 with a limit of 2. Then does ipv6 also use "You are being redirected to ..." for the 10 items per ASN? So the max allocated ip per ASN is 12 including ipv4 and ipv6?

Reply:
> So the max allocated ips per ASN is actually 12 including ipv4 and ipv6?

Yes. The maximum number of seeds per ASN is 12: 2 IPv4 and 10 IPv6.

Reply:
> Which asmap are you using? It seems more likely to me that your asmap is wrong, or something is wrong with parsing, causing all IPs to be assigned to one AS, than that this change is at fault.

Good call; fixed in the latest push.